### PR TITLE
[module_test_ios] trying tap the buttons again if the first time didn't work

### PR DESCRIPTION
**Pull request by cyanglaz**
I guess there's a race condition in the XCUITests that caused some flakiness in the tests. This PR is to try another workaround, the [first try](https://github.com/flutter/flutter/pull/90149) with tap on app didn't work. 

https://github.com/flutter/flutter/issues/89175

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/integration_tests/ios_host_app/FlutterUITests/FlutterUITests.m [hunk 1]
```diff
@@ -36,7 +36,12 @@ - (void)testFullScreenWarm {
     XCUIApplication *app = self.app;
 
     [self waitForAndTapElement:app.buttons[@"Full Screen (Warm)"]];
-    XCTAssertTrue([app.staticTexts[@"Button tapped 0 times."] waitForExistenceWithTimeout:60.0]);
+    BOOL newPageAppeared = [app.staticTexts[@"Button tapped 0 times."] waitForExistenceWithTimeout:60.0];
```
Comment (jmagman) [inline dev/integration_tests/ios_host_app/FlutterUITests/FlutterUITests.m:39]: I suspect 60 seconds is too long for everything in this test, but we can reevaluate once this test is stable.

File: dev/integration_tests/ios_host_app/FlutterUITests/FlutterUITests.m [hunk 2]
```diff
@@ -36,7 +36,12 @@ - (void)testFullScreenWarm {
...
+    if (!newPageAppeared) {
```
[the remark this answers is not in the document]
Comment (cyanglaz) [reply dev/integration_tests/ios_host_app/FlutterUITests/FlutterUITests.m:40]: done

File: dev/integration_tests/ios_host_app/FlutterUITests/FlutterUITests.m (hunk 1 above)
Comment (cyanglaz) [reply dev/integration_tests/ios_host_app/FlutterUITests/FlutterUITests.m]: I refactored a little so all the 60.0 are now a constant, and yeah we can adjust it later.
Note that there are 2 time-outs in this tests have 30.0 time outs

**Review by jmagman — APPROVED** [review]
LGTM
🤞

**Comment by jmagman** [comment]
You can rebase on master to clear up the tool test errors.

**Comment by cyanglaz** [comment]
looks like the google testing is stuck (it's been over 24 hours), I'll re-trigger CI

**Pull request merged**